project/src/js/ui/inbox/compose/composeView.tsx

```diff
@@ -0,0 +1,68 @@
+import * as h from '../../helper';
+import { header as headerWidget, backButton } from '../../shared/common';
+import layout from '../../layout';
+import i18n from '../../../i18n';
+import { ComposeAttrs, ComposeState } from '../interfaces';
+
+export default function view(vnode: Mithril.Vnode<ComposeAttrs>) {
+  const ctrl = vnode.state as ComposeState;
+  const headerCtrl = () => headerWidget(null,
+    backButton(i18n('composeMessage'))
+  );
+  const bodyCtrl = composeBody.bind(undefined, ctrl);
+  return layout.free(headerCtrl, bodyCtrl, undefined, undefined);
+}
+
+function composeBody(ctrl: ComposeState) {
+  return (
+    <div className="composeWrapper">
+      <form id="composeForm"
+      onsubmit={function(e: Event) {
+        e.preventDefault();
+        return ctrl.send(e.target as HTMLFormElement);
+      }}>
+        {ctrl.id() ? recipientWithName(ctrl) : recipientWithoutName()}
+        <div key="recipientError" className="errorMessage">
```

Comment:
Here we have empty divs when there is no error, why not render conditionally the div as well?
You'll need to ensure every element in `<form>` has a unique key.

Idem for other error fields